### PR TITLE
minor updates to the substrate doc crate

### PR DESCRIPTION
**Pull request by kianenigma**

### CODE REVIEW

**Review by liamaharon — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -183,7 +201,7 @@
 //! [`node-template`]: ../node_template/index.html
 //! [`kitchensink_runtime`]: ../kitchensink_runtime/index.html
 //! [`subkey`]: ..//subkey/index.html
```
Comment (liamaharon) [inline src/lib.rs:203]:
```suggestion
//! [`subkey`]: ../subkey/index.html
```

**Review by bkchr — APPROVED** [review]

**Review by altaua — APPROVED** [review]

**Comment by kianenigma** [comment]
bot fmt

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3242093

File: src/lib.rs [hunk 2]
```diff
@@ -105,6 +105,21 @@
 //! * `pallet-*` and `frame-*` crates, located under `./frame` folder. These are the crates related
 //!   to FRAME. See [`frame_support`] for more information.
 //!
+//! ### Wasm Build
+//!
+//! Many of the Substrate crates, such as entire `sp-*`, need to compile to both Wasm (when a Wasm
+//! runtime is being generated) and native ! (for example, when testing). To achieve this, Substrate
```
Comment (ggwpez) [inline src/lib.rs:111]:
```suggestion
//! runtime is being generated) and native (for example, when testing). To achieve this, Substrate
```

File: src/lib.rs [hunk 3]
```diff
@@ -119,6 +134,22 @@
...
+//! From the above, [`node`] and [`node-template`] are essentially blueprints of a substrate-based
+//! project, as the name of the latter is implying. Each substrate-based project typically contains
+//! the following:
+//!
+//! * Under `./runtime`, a `./runtime/src/lib.rs` which is the top level runtime amalgamator file.
+//!   This file typically contains the [`frame_support::construct_runtime`] macro, which is the
+//!   final definition of a runtime.
+//!
+//! * Under `./node`, a `main.rs`, which is the point, and a `./service.rs`, which contains all the
+//!   client side components. Skimming this file yield an overview of the networking, database,
```
Comment (ggwpez) [inline src/lib.rs:148]:
```suggestion
//!   client side components. Skimming this file yields an overview of the networking, database,
```

**Review by ggwpez — APPROVED** [review]

**Review by lovelaced — APPROVED** [review]

**Pull request merged**